Makefile.toml

```diff
@@ -1352,5 +1355,106 @@ rm -rf ${BUILDSYS_STATE_DIR}
 '''
 ]
 
+[tasks.test-tools]
+dependencies = ["setup", "fetch-sources"]
+script = [
+    '''
+    cargo install \
+      ${CARGO_MAKE_CARGO_ARGS} \
+      --path tools/testsys \
+      --root tools \
+      --force \
+      --quiet
+    '''
+]
+
+[tasks.test-setup]
+dependencies = ["test-tools"]
+script = [
+    '''
+    set -eu
+    export PATH="${BUILDSYS_TOOLS_DIR}/bin:${PATH}"
+    testsys install
+    '''
+]
+
+# This task is used to test bottlerocket build artifacts. By default the region first listed in Infra.toml
+# is used for testing; however, `TESTSYS_REGION` can be used to test in a different region.
+[tasks.test]
+dependencies = ["test-tools"]
+script = [
+    '''
+    set -eu
+    ami_input="${BUILDSYS_OUTPUT_DIR}/${BUILDSYS_NAME_FULL}-${AMI_DATA_FILE_SUFFIX}"
+    if [ ! -s "${ami_input}" ]; then
+      echo "AMI input file doesn't exist for the current version/commit - ${BUILDSYS_VERSION_FULL} - please run 'cargo make ami'" >&2
+      exit 1
+    fi
+    export PATH="${BUILDSYS_TOOLS_DIR}/bin:${PATH}"
+    # The ami that is selected from `amis.json` is determined by `TESTSYS_REGION` if set; otherwise,
+    # it is the first region listed in `Infra.toml` (for aws variants).
+    testsys run ${TESTSYS_TEST} \
+      --ami-input "${ami_input}" \
+      ${TESTSYS_AWS_SECRET_NAME:+--secret ${TESTSYS_AWS_SECRET_NAME}} \
+    '''
+]
+
+# This task will clear all tests and resources from the testsys cluster.
+[tasks.clean-test]
+dependencies = ["test-tools"]
+script = [
+    '''
+    set -eu
+    export PATH="${BUILDSYS_TOOLS_DIR}/bin:${PATH}"
+    testsys delete
+    '''
+]
+
+# This task will clear all testsys components from the testsys cluster.
+[tasks.uninstall-test]
+dependencies = ["test-tools"]
+script = [
+   '''
+   set -eu
+   # Replace with `testsys uninstall`s once
+   # https://github.com/bottlerocket-os/bottlerocket-test-system/pull/450 is merged
+   kubectl delete namespace testsys-bottlerocket-aws
```

Comment:
How are we ensuring that the caller has the right kubeconfig context set up?

Reply:
Because the caller is required to have the right Kubeconfig set up for all of the testsys commands/targets.